Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -35,32 +35,110 @@ import TensorFlow
 /// [2]: https://github.com/borglab/gtsam/blob/develop/doc/math.pdf
 /// [3]: Actually, we define the pullbacks because Swift doesn't support differentials very well
 ///      yet.
-public struct Pose2: Differentiable, TangentStandardBasis, KeyPathIterable {
-  /// The pose's rotation.
-  ///
-  /// This is the projection SE(2) -> SO(2).
-  @differentiable public var rot: Rot2 { rotStorage }
+public struct Pose2: Manifold, Equatable, TangentStandardBasis, KeyPathIterable {
+  // MARK: - Manifold conformance
 
-  /// The pose's translation.
-  ///
-  /// This is the projection SE(2) -> R^2. (Note: not a group homomorphism!)
-  @differentiable public var t: Vector2 { tStorage }
+  public var coordinateStorage: Pose2Coordinate
+  public init(coordinateStorage: Pose2Coordinate) { self.coordinateStorage = coordinateStorage }
+
+  public mutating func move(along direction: Coordinate.LocalCoordinate) {
+    coordinateStorage = coordinateStorage.global(direction)
+  }
 
   /// Creates a `Pose2` with rotation `r` and translation `t`.
   ///
   /// This is the bijection SO(2) x R^2 -> SE(2), where "x" means direct product of groups. (Note:
   /// not a group homomorphism!)
   @differentiable
   public init(_ r: Rot2, _ t: Vector2) {
-    tStorage = t
-    rotStorage = r
+    self.init(coordinate: Pose2Coordinate(r, t))
+  }
+  
+  // MARK: Convenience Attributes
+  
+  @differentiable public var t: Vector2 { coordinate.t }
+  
+  @differentiable public var rot: Rot2 { coordinate.rot }
+  
+  /// Twist coordinates, with the first component for rotation and the remaining components for
+  /// translation.
+  ///
+  /// See the documentation comment on the declaration of `Pose2` for more information.
+  public typealias TangentVector = Vector3
+
+  /// Product of two rotations.
+  @differentiable
+  public static func * (lhs: Pose2, rhs: Pose2) -> Pose2 {
+    Pose2(coordinate: lhs.coordinate * rhs.coordinate)
+  }
+  
+  @differentiable
+  public func inverse() -> Pose2 {
+    Pose2(coordinate: coordinate.inverse())
+  }
+}
+
+// MARK: Convenience initializers
+extension Pose2 {
+  /// Creates a `Pose2` with translation `x` and `y` and with rotation `theta`.
+  @differentiable
+  public init(_ x: Double, _ y: Double, _ theta: Double) {
+    self.init(Rot2(theta), Vector2(x, y))
+  }
+
+  public init(randomWithCovariance covariance: Tensor<Double>) {
+    self.init(0, 0, 0)
+    let r = matmul(cholesky(covariance), Tensor<Double>(randomNormal: [3, 1])).scalars
+    let tv = Pose2.TangentVector(r[0], r[1], r[2])
+    self.move(along: tv)
+  }
+}
+
+// MARK: - Global Coordinate System
+
+public struct Pose2Coordinate: Equatable, KeyPathIterable {
+  var tStorage: Vector2
```

Comment:
There is no need to have stored properties `tStorage` and `rotStorate` with wrappers `t` and `rot`.

Instead, you can just have stored properties `t: Vector2` and `rot: Rot2`.